addons/interact_menu/functions/fnc_removeActionFromClass.sqf

```diff
@@ -6,18 +6,29 @@
  * 0: TypeOf of the class <STRING>
  * 1: Type of action, 0 for actions, 1 for self-actions <NUMBER>
  * 2: Full path of the new action <ARRAY>
+ * 3: Allow inheritance <BOOL> (default: false)
  *
  * Return Value:
  * None
  *
  * Example:
- * [typeOf cursorTarget, 0,["ACE_TapShoulderRight","VulcanPinch"]] call ace_interact_menu_fnc_removeActionFromClass;
+ * [typeOf cursorTarget, 0,["ACE_TapShoulderRight","VulcanPinch"], true] call ace_interact_menu_fnc_removeActionFromClass;
  *
  * Public: No
  */
 #include "script_component.hpp"
 
-params ["_objectType", "_typeNum", "_fullPath"];
+params ["_objectType", "_typeNum", "_fullPath", ["_inherit", false, [false]]];
+
+// Call this function for all child classes if inheritance is enabled.
+if (_inherit) then {
+    private _children = (
+        (format ["_class = configName _x; (_class isKindOf '%1') && (_class != '%1')", _objectType]) configClasses (configFile >> "CfgVehicles")
```

Comment:
it will be faster and pretty to check condition in count/forEach loop for `'true' configClasses` array

Reply:
Why are two loops faster than one loop?

Reply:
Is this better?
```
if (_inherit) then {
    {
        private _class = configName _x;
        if ((_class isKindOf _objectType) && {_class != _objectType}) then {
            [_class, _typeNum, _fullPath] call FUNC(removeActionFromClass);
        };
    } forEach ("true" configClasses (configFile >> "CfgVehicles"));
};
```

Reply:
Nope way worse. Adding a if and then for each element. And as mentioned the second loop
Actually looking at the current code. you have 3 loops but you only two.
```
private _children = (format ["_class = configName _x; (_class isKindOf '%1') && (_class != '%1')", _objectType]) configClasses (configFile >> "CfgVehicles");
_children apply { [configName _x, _typeNum, _fullPath] call FUNC(removeActionFromClass) }
```
Instead of apply you could use any loop type you want. count/forEach/select/apply 
But I'd say apply is the most readable of all.
count might be faster though if removeActionFromClass returns nil.

Reply:
@dedmen yeah, looks like BI optimized `configClasses` much, it was much slower months ago.
@xrufix sorry for bothering

Reply:
It should be better now.